app/main/views/frameworks.py

```diff
@@ -332,7 +324,17 @@ def download_agreement_file(framework_slug, document_name):
         abort(404)
 
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    path = get_agreement_document_path(framework_slug, current_user.supplier_id, document_name)
+
+    if document_name == 'signed-agreement':
```

Comment:
Sorry for duplication - obviously I meant that we need to retain the old behaviour for *unsigned* agreement downloads.